packages/core/src/parser/cstToAstTransformer.ts

```diff
@@ -343,18 +397,11 @@ export function generateCstToAstTransfromer(parser: Parser): ICstVisitor<never,
          * @returns the generated FieldAccessExpression
          */
         private generateFieldAccessExpressionForOnlyDot(baseExpression: Expression, dotToken: IToken): Expression {
-            const dotPosition = generatePosition(dotToken, dotToken);
+            const dotRange = generateRange(dotToken, dotToken);
             return new FieldAccessExpression("", baseExpression, {
-                ...generateMetadata(baseExpression.position, dotToken),
-                completionPosition: dotPosition,
-                identifierPosition: {
-                    startOffset: dotPosition.endOffset,
-                    startLine: dotPosition.endLine,
-                    startColumn: dotPosition.endColumn,
-                    endOffset: dotPosition.endOffset,
-                    endLine: dotPosition.endLine,
-                    endColumn: dotPosition.endColumn
-                }
+                ...generateMetadata(baseExpression.range, dotToken),
+                completionRange: dotRange,
+                identifierRange: [dotRange[1], dotRange[1]]
```

Comment:
ups, overlooked this one
These two have different semantics, regarding how completion works.
The `identifierRange` is the part which is replaced when you select a completion item from the list. In case you just typed a dot, there is nothing to replace, you just insert directly after the dot (thus this range of length 0)
The `completionRange` is a bit more complicated.
When a completion is triggered, we execute the program, and stop when we reach the point where we want to have completion. This range "marks" where this expression would be responsible for capturing the context. To be precise, it is responsible if the cursor is either inside the range (but not at the start) or immediately after the range.
In this case, this causes the completion to be triggered when you want to have a completion immediately after the dot